Sources/Load.php

```diff
@@ -64,12 +64,28 @@ function reloadSettings()
 
 		// We explicitly do not use $smcFunc['json_decode'] here yet, as $smcFunc is not fully loaded.
 		if (!is_array($modSettings['attachmentUploadDir']))
-			$modSettings['attachmentUploadDir'] = smf_json_decode($modSettings['attachmentUploadDir'], true);
+		{
+			$attachmentUploadDir = smf_json_decode($modSettings['attachmentUploadDir'], true);
+			$modSettings['attachmentUploadDir'] = !empty($attachmentUploadDir) ? $attachmentUploadDir : $modSettings['attachmentUploadDir'];
+		}
 
 		if (!empty($cache_enable))
 			cache_put_data('modSettings', $modSettings, 90);
 	}
 
+	// Going anything further when the files don't match the database can make nasty messes (unless we're actively installing or upgrading)
+	if (!defined('SMF_INSTALLING') && !empty($modSettings['smfVersion']) && version_compare(strtolower(strtr($modSettings['smfVersion'], array('SMF ' => '', ' ' => '.'))), strtolower(strtr($forum_version, array('SMF ' => '', ' ' => '.'))), '!='))
+	{
```

Comment:
How would this work with the ```$forum_verison = '2.1.1';``` and ```$modSettings['smfVersion'] = '2.1.0';```.  Such in the case of upgrading to a minor patch version?

Reply:
```
<?php

$forum_version = 'SMF 2.1.1';
$modSettings['smfVersion'] = '2.1.0';

if (!empty($modSettings['smfVersion']) && version_compare(strtolower(strtr($modSettings['smfVersion'], array('SMF ' => '', ' ' => '.'))), strtolower(strtr($forum_version, array('SMF ' => '', ' ' => '.'))), '!='))
	echo 'different';
else
	echo 'same';
```

Result: 'different'

Reply:
So, patches will always need to update both the DB value and the string in index.php if they update either. But that's just good practice anyway.